sdc/datatypes/hpat_pandas_series_functions.py

```diff
@@ -45,12 +49,114 @@
                                             hpat_join_series_indexes)
 from sdc.datatypes.hpat_pandas_series_rolling_types import _hpat_pandas_series_rolling_init
 from sdc.datatypes.hpat_pandas_stringmethods_types import StringMethodsType
+from sdc.datatypes.hpat_pandas_getitem_types import SeriesGetitemAccessorType
 from sdc.hiframes.pd_series_ext import SeriesType
 from sdc.str_arr_ext import (StringArrayType, cp_str_list_to_array, num_total_chars, string_array_type,
                              str_arr_is_na, pre_alloc_string_array, str_arr_set_na)
 from sdc.utils import to_array, sdc_overload, sdc_overload_method, sdc_overload_attribute
 
 
+@sdc_overload(operator.getitem)
+def hpat_pandas_series_accessor_getitem(self, idx):
+    """
+    Pandas Series operator :attr:`pandas.Series.get` implementation
+    **Algorithm**: result = series[idx]
+
+    **Test**: python -m sdc.runtests sdc.tests.test_series.TestSeries.test_static_getitem_series1
+
+    Parameters
+    ----------
+    series: :obj:`pandas.Series`
+           input series
+    idx: :obj:`int`, :obj:`slice` or :obj:`pandas.Series`
+        input index
+
+    Returns
+    -------
+    :class:`pandas.Series` or an element of the underneath type
+            object of :class:`pandas.Series`
+    """
+
+    _func_name = 'Operator getitem().'
+
+    if not isinstance(self, SeriesGetitemAccessorType):
+        return None
+
+    accessor = self.accessor.literal_value
+
+    if accessor == 'iloc':
+        if isinstance(idx, types.SliceType):
+            def hpat_pandas_series_iloc_slice_impl(self, idx):
+                return pandas.Series(self._series._data[idx])
+
+            return hpat_pandas_series_iloc_slice_impl
+
+        def hpat_pandas_series_iloc_impl(self, idx):
+            return self._series._data[idx]
+
+        return hpat_pandas_series_iloc_impl
+
+    if accessor == 'iat':
+        if isinstance(idx, (int, types.Integer)):
+            def hpat_pandas_series_iat_impl(self, idx):
+                return self._series._data[idx]
+
+            return hpat_pandas_series_iat_impl
+
+        raise TypingError('{} The index must be a Integer. Given: {}'.format(_func_name, idx))
+
+    if accessor == 'loc':
+    #note: Loc return Series
+        index_is_none = (self.series.index is None or
+                         isinstance(self.series.index, numba.types.misc.NoneType))
+        if isinstance(idx, types.SliceType) and not index_is_none:
+            def hpat_pandas_series_getitem_idx_slice_impl(self, idx):
+                start = -1
+                stop = -1
+                for i in numba.prange(len(self._series._index)):
+                    if self._series._index[i] == idx.start and start == -1:
+                        start = i
+                    if self._series._index[i] == idx.stop and start != -1 and stop == -1:
+                        stop = i 
+                return pandas.Series(self._series._data[start:stop+1], self._series._index[start:stop+1])
```

Comment:
It seems to me that we should write a restriction for users that the slice supports only the specified start and end. Because if they are not specified, start = 0 and end = 9223372036854775807 come to the registered function, in the case of non-standard indexes we cannot distinguish the data that the user set from the default

Reply:
How pandas behave in such cases?

Reply:
9223372036854775807 + 1 is zero. So, in case of stop == 9223372036854775807 you souldn't do +1